components/kyma-environment-broker/internal/broker/instance_get_test.go

```diff
@@ -90,6 +90,42 @@ func TestGetEndpoint_GetProvisioningInstance(t *testing.T) {
 	assert.Len(t, response.Metadata.Labels, 2)
 }
 
+func TestGetEndpoint_GetInstanceWhereDeletedAtIsNotZero(t *testing.T) {
+	// given
+	st := storage.NewMemoryStorage()
+	cfg := broker.Config{
+		URL:                                     "https://test-broker.local",
+		EnableKubeconfigURLLabel:                true,
+		ShowTrialExpirationInfo:                 true,
+		SubaccountsIdsToShowTrialExpirationInfo: "test-saID",
+	}
+
+	const (
+		instanceID  = "cluster-test"
+		operationID = "operationID"
+	)
+	op := fixture.FixProvisioningOperation(operationID, instanceID)
+
+	instance := fixture.FixInstance(instanceID)
+	instance.DeletedAt = time.Now()
+
+	err := st.Operations().InsertOperation(op)
+	require.NoError(t, err)
+
+	err = st.Instances().Insert(instance)
+	require.NoError(t, err)
+
+	svc := broker.NewGetInstance(cfg, st.Instances(), st.Operations(), logrus.New())
+
+	// when
+	_, err = svc.GetInstance(context.Background(), instanceID, domain.FetchInstanceDetails{})
+
+	// then
+	assert.IsType(t, err, &apiresponses.FailureResponse{}, "Get returned error of unexpected type")
```

Comment:
Kind of misleading message. It is not so obvious that `Get` means GET or Get request.

Reply:
And why we got here `unexpected type`?

Reply:
The messages come from `TestGetEndpoint_GetNonExistingInstance` since we expect similar result in both tests that the instance doesn't exist.

Reply:
Got it.